PR: https://github.com/odoo/odoo/pull/218612

From: 45c2f90bc377fc2576572f6750c53647c6af7142
From: dhha-odoo
Date: 2026-01-14 15:10:33

Breaking data model changes scores: del:3 + add:0, change matches:
-    on_time_rate_perc = fields.Float(string="OTD", compute="_compute_on_time_rate_perc")
-    _inherit = 'purchase.order.line'
-    on_time_rate_perc = fields.Float(string="OTD", related="order_id.on_time_rate_perc")

Total Changes: 19

[MOV] purchase_alternative_stock*: separate purchase_requisition_stock

*purchase_requisition_stock

With This Commit:
=================================

We have separated the purchase alternative–related functionality from the
purchase_requisition_stock module and reorganised it under a newly created module, purchase_alternative_stock.
This new module depends on both purchase_alternative and purchase_stock,
ensuring proper modular structure and clearer separation of responsibilities.

taskId :- 4737059

Part-of: odoo/odoo#218612
Related: odoo/upgrade#8052
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/purchase_requisition_stock/models/purchase.py
+++ b/addons/purchase_requisition_stock/models/purchase.py
@@ -1,30 +1,13 @@
-# -*- encoding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import api, models
 
 
 class PurchaseOrder(models.Model):
     _inherit = 'purchase.order'
 
-    on_time_rate_perc = fields.Float(string="OTD", compute="_compute_on_time_rate_perc")
-
-    @api.depends('on_time_rate')
-    def _compute_on_time_rate_perc(self):
-        for po in self:
-            if po.on_time_rate >= 0:
-                po.on_time_rate_perc = po.on_time_rate / 100
-            else:
-                po.on_time_rate_perc = -1
-
     @api.onchange('requisition_id')
     def _onchange_requisition_id(self):
         super(PurchaseOrder, self)._onchange_requisition_id()
         if self.requisition_id:
             self.picking_type_id = self.requisition_id.picking_type_id.id
-
-
-class PurchaseOrderLine(models.Model):
-    _inherit = 'purchase.order.line'
-
-    on_time_rate_perc = fields.Float(string="OTD", related="order_id.on_time_rate_perc")
